### PR TITLE
nanocoap_sock: always use coap_opt_put_uri_pathquery()

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

This is pretty subtle: `coap_opt_put_uri_pathquery()` will detect the presense of a query parameter, `coap_opt_put_uri_path()` will not.

That means that when you request `path/to/resource?option=1` with `coap_opt_put_uri_path()` you get the path `resource?option=1` whereas with `coap_opt_put_uri_pathquery()` you get the path `resource` with query parameter `option=1`.

We really want the latter.

### CODE REVIEW

File: sys/net/application_layer/nanocoap/sock.c [hunk 1]
```diff
@@ -384,7 +384,7 @@ ssize_t nanocoap_sock_get(nanocoap_sock_t *sock, const char *path, void *buf, si
 
     pktpos += coap_build_hdr(pkt.hdr, COAP_TYPE_CON, NULL, 0, COAP_METHOD_GET,
                              nanocoap_sock_next_msg_id(sock));
-    pktpos += coap_opt_put_uri_path(pktpos, 0, path);
+    pktpos += coap_opt_put_uri_pathquery(pktpos, 0, path);
```
Comment (chrysn) [inline sys/net/application_layer/nanocoap/sock.c:387]:
```suggestion
    pktpos += coap_opt_put_uri_pathquery(pktpos, NULL, path);
```

File: sys/net/application_layer/nanocoap/sock.c [hunk 2]
```diff
@@ -533,7 +534,7 @@ ssize_t nanocoap_sock_delete(nanocoap_sock_t *sock, const char *path)
 
     pktpos += coap_build_hdr(pkt.hdr, COAP_TYPE_CON, NULL, 0, COAP_METHOD_DELETE,
                              nanocoap_sock_next_msg_id(sock));
-    pktpos += coap_opt_put_uri_path(pktpos, 0, path);
+    pktpos += coap_opt_put_uri_pathquery(pktpos, 0, path);
```
Comment (chrysn) [inline sys/net/application_layer/nanocoap/sock.c:537]:
```suggestion
    pktpos += coap_opt_put_uri_pathquery(pktpos, NULL, path);
```

**Review by chrysn — COMMENTED** [review]
LGTM in the sense that it is probably what people need and looks correct from an implementation PoV.

The docs will need updating in all the functions like nanocoap_sock_get that previously speak of a "remote path" to now say "remote path and query".

And it is an API change, moreover, one that is not trivially raising compiler warnings. Given that it's likely reflecting any user's intention, it should suffice to add the right labels to make sure it gets a prominent note in the change log.

**Comment by benpicco** [comment]
To me it feels more like a bug fix than an API change - when I added the functions I was not aware of the difference between `coap_opt_put_uri_pathquery()` and `coap_opt_put_uri_path()` (or that there are in fact two different functions) :sweat_smile:

**Comment by benpicco** [comment]
Is there anything I can do to move this forward?

**Review by maribu — APPROVED** [review]

**Comment by benpicco** [comment]
Thank you!

**Pull request merged**